src/services/user.service.js

```diff
@@ -0,0 +1,55 @@
+let users = [];
+let nextId = 1;
+
+const resetUsers = () => {
+  users = [];
+  nextId = 1;
+};
+
+const getAll = () => {
+  return users;
+};
+
+const getById = (id) => {
+  const userId = Number(id);
+
+  return users.find((person) => person.id === userId) || null;
+};
+
+const create = (name) => {
+  const newUser = {
+    id: nextId,
+    name,
+  };
+
+  nextId++;
+
+  users.push(newUser);
+
+  return newUser;
+};
+
+const remove = (id) => {
+  const filteredUsers = users.filter((person) => person.id !== id);
+
+  users = filteredUsers;
+
+  return users;
```

Comment:
The `remove` function returns the entire `users` array, which might not be necessary. Consider returning a boolean to indicate success or failure instead.